jenkins/PerfCI-Chaos/01-PerfCI-Chaos-OpenShift-Deployment/Jenkinsfile

```diff
@@ -0,0 +1,175 @@
+pipeline {
+    agent {
+        label 'PerfCi'
+    }
+    environment {
+        INSTALL_OCP_VERSION = '4.14.35'
+        CONTACT_EMAIL1 = credentials('perfci_contact_email1')
+        PROVISION_PRIVATE_KEY_FILE = credentials('perfci_provision_private_key_file')
+        PROVISION_IP = credentials('perfci_provision_ip')
+        PROVISION_USER = credentials('perfci_provision_user')
+        CLUSTER_TYPE = credentials('perfci_cluster_type')
+        EXPECTED_NODES = credentials('perfci_expected_nodes')
+        OCP_CLIENT_VERSION = credentials('perfci_ocp_client_version')
+        QUAY_BENCHMARK_RUNNER_REPOSITORY = credentials('perfci_quay_repository')
+        WORKSPACE = '/home/jenkins'
+        PROVISION_PORT = 22
+        KUBEADMIN_PASSWORD_PATH = '/root/.kube/kubeadmin-password'
+        KUBECONFIG_PATH = '/root/.kube/config'
+        PROVISION_INSTALLER_PATH = '/root/jetlag/./run_jetlag.sh'
+        PROVISION_INSTALLER_CMD = 'pushd /root/jetlag;/root/jetlag/./run_jetlag.sh 1>/dev/null 2>&1;popd'
+        PROVISION_INSTALLER_LOG = 'tail -100 /root/jetlag/jetlag.log'
+        PRIVATE_KEY_PATH = '/home/jenkins/.ssh/provision_private_key'
+        CONFIG_PATH = '/home/jenkins/.ssh/config'
+        CONTAINER_PRIVATE_KEY_PATH = '/root/.ssh/provision_private_key'
+        CONTAINER_CONFIG_PATH = '/root/.ssh/config'
+        ANSIBLE_TMP_PATH = '/root/.ansible/tmp/'
+        PROVISION_TIMEOUT = 7200
+    }
+    stages {
+        stage('Cleanup') {
+            steps {
+                script {
+                    try {
+                        // Clean the Jenkins workspace
+                        echo "Cleaning Jenkins workspace"
+                        deleteDir()
+
+                        // Delete assisted installer previous deployments
+                        sh "curl http://${PROVISION_IP}:8090/api/assisted-install/v2/clusters | jq '.[].id' -r | xargs -I % curl -X DELETE http://${PROVISION_IP}:8090/api/assisted-install/v2/clusters/%"
+
+                        // remove ansible tmp files
+                        sh "sudo rm -rf ${ANSIBLE_TMP_PATH}"
+
+                        // Delete benchmark-runner images
+                        sh '''
+                        if [[ "$(sudo podman images -q ${QUAY_BENCHMARK_RUNNER_REPOSITORY} 2> /dev/null)" != "" ]]; then
```

Comment:
Make sure that there are no running containers using any of these images, or the rmi -f will fail.

Reply:
The main purpose of this deletion is to remove all running containers of benchmark-runner to allow a fresh deployment.

Reply:
Right, but I don't think `podman rmi` removes images for running containers.